src/huggingface_hub/hf_api.py

```diff
@@ -1355,8 +1381,10 @@ def move_repo(
             "Accepted transfer request. You will get an email once this is successfully completed."
         )
 
+    @_deprecate_positional_args
     def upload_file(
         self,
+        *,
         path_or_fileobj: Union[str, bytes, IO],
         path_in_repo: str,
```

Comment:
I'd personally put it right after the `path_or_fileobj`:
```suggestion
    def upload_file(
        self,
        path_or_fileobj: Union[str, bytes, IO],
        *,
        path_in_repo: str,
```

Reply:
I kinda would agree, but there is 0 instance of `upload_file` usage in the repo which passes this arg positionally.